Cargo.toml

```diff
@@ -80,6 +80,7 @@ url = "2.2"
 bytes = "1.0"
 serde = "1.0"
 serde_urlencoded = "0.7"
+itertools = "0.10.0"
```

Comment:
Doesn't reqwest already use the `Either` from futures? I think it'd be worth trying not to add the extra dependency just for `Either`.

Reply:
reqwest uses the `Either` type from futures_util. [This does not impl Iterator](https://docs.rs/futures-util/0.3.15/futures_util/future/enum.Either.html) in the case when both inner types do. I will get a new version up with a simple iterator Either implemented.